room/compiler-processing/src/main/java/androidx/room/compiler/processing/javac/JavacProcessingEnv.kt

```diff
@@ -183,6 +188,69 @@ internal class JavacProcessingEnv(
         } as T
     }
 
+    internal inline fun wrapElement(
```

Comment:
passing annotation here feels very weird.
I see you are passing it in to have a better error message.

I have two possible options:

a) rename method to `wrapAnnotatedElement` and also pass the annotationName as a parameter (rather then callback)
b) define an explict exception and catch it in the call site to append the annotation name to the user visible error.

I'm equal on both options so feel free to pick one or a third one.

Reply:
Agreed, I can do a) in a follow up